src/tcp-socket.js

```diff
@@ -73,11 +72,214 @@
     } else if (typeof chrome !== 'undefined' && (chrome.socket || chrome.sockets)) {
         // chrome packaged app using chrome.socket
         chromeShim();
+    } else if (typeof Windows === 'object' && Windows) {
```

Comment:
We might want to check for the `Windows.Networking.Sockets.StreamSocket` object here just to be sure. Don't know if the `Windows` object is there in other runtimes.